### PR TITLE
Remediate critical SonarCloud bugs AB#15494

### PR DESCRIPTION
**Pull request by MikeLyttle**
# Fixes [AB#15494](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/15494)

## Description

- updates Vue getters to always explicitly return
- reduces cognitive complexity

## Testing

- [ ] Unit Tests Updated
- [ ] Functional Tests Updated
- [x] Not Required

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

**Review by BrianMaki — APPROVED** [review]

File: Apps/JobScheduler/src/Jobs/NotificationSettingsJob.cs [hunk 1]
```diff
@@ -74,51 +74,47 @@ public class NotificationSettingsJob : INotificationSettingsJob
         public void PushNotificationSettings(string notificationSettingsJson)
         {
             this.logger.LogDebug("Queueing Notification Settings push to PHSA...");
-            if (this.jobEnabled)
+            if (!this.jobEnabled)
```
Comment (ytqsl) [inline Apps/JobScheduler/src/Jobs/NotificationSettingsJob.cs:77]: This is quite dangerous - if a job is disabled, it should be disabled from being scheduled by Hangfire

File: Apps/JobScheduler/src/Jobs/NotificationSettingsJob.cs [hunk 2]
```diff
@@ -74,51 +74,47 @@ public class NotificationSettingsJob : INotificationSettingsJob
...
-                    if (string.IsNullOrEmpty(accessToken))
-                    {
-                        this.logger.LogError("Authenticated as User System access token is null or empty, Error:\n{AccessToken}", accessToken);
-                        throw new FormatException($"Authenticated as User System access token is null or empty, Error:\n{accessToken}");
-                    }
+            NotificationSettingsRequest? notificationSettings = JsonSerializer.Deserialize<NotificationSettingsRequest>(notificationSettingsJson);
+            if (notificationSettings == null)
+            {
+                this.logger.LogError("Unable to deserialize JSON Notification Settings");
+                throw new FormatException("Unable to deserialize JSON Notification Settings");
```
Comment (ytqsl) [inline Apps/JobScheduler/src/Jobs/NotificationSettingsJob.cs:87]: Not fond of using FormatException here - InvalidOperationException might be more sensible, however, it was there before :)

File: Apps/AccountDataAccess/src/Patient/PatientRepository.cs
Comment (ytqsl) [inline Apps/AccountDataAccess/src/Patient/PatientRepository.cs:1]: I think this method deserves more attention in the future, @BrianMaki  and I discussed refactoring to a strategy pattern (out of scope for now)

**Review by ytqsl — APPROVED** [review]
some comments that shouldn't block from merging

**Review by BronzBierd — APPROVED** [review]
Looks good.

File: Apps/JobScheduler/src/Jobs/NotificationSettingsJob.cs (hunk 1 above)
Comment (MikeLyttle) [reply Apps/JobScheduler/src/Jobs/NotificationSettingsJob.cs:77]: I made no change to the logic here, just inverted the if statements to reduce indentation, which reduced cognitive complexity.

**Pull request merged**